static/js/stream_popover.js

```diff
@@ -500,6 +512,38 @@ exports.register_topic_handlers = function () {
 
         e.stopPropagation();
     });
+
+    $('body').on('click', '.sidebar-popover-move-topic-messages', function (e) {
+        const topic_row = $(e.currentTarget);
+        const stream_id = topic_row.attr('data-stream-id');
+        const topic_name = topic_row.attr('data-topic-name');
+        const message = home_msg_list.all_messages()
+            .find(m => m.topic === topic_name && m.stream_id === +stream_id);
+        build_move_topic_to_stream_popover(e, message.id, stream_id, topic_name);
+        e.stopPropagation();
+        e.preventDefault();
+    });
+
+    $('body').on('click', '#do_move_topic_button', function (e) {
+        const params = $('#move_topic_form').serializeArray().reduce(function (obj, item) {
+            obj[item.name] = item.value;
+            return obj;
+        }, {});
+
+        const {message_id, select_stream_id, old_topic_name} = params;
+        let {topic_name} = params;
+
+        if (old_topic_name.trim() === topic_name.trim()) {
+            topic_name = undefined;
+        }
```

Comment:
Maybe add a comment here noting that this is intended to request no change in the topic if it didn't change?

Reply:
Done